Tests/Library/ConnectionStringParser.tests.ts

```diff
@@ -0,0 +1,104 @@
+import assert = require("assert");
+
+import sinon = require("sinon");
+import Constants = require("../../Declarations/Constants");
+import ConnectionStringParser = require("../../Library/ConnectionStringParser");
+
+describe("ConnectionStringParser", () => {
+    describe("#parse()", () => {
+        it("should parse all valid fields", () => {
+            const authorization = "ikey"
+            const instrumentationKey = "instr_key";
+            const ingestionEndpoint = "ingest";
+            const liveEndpoint = "live";
+            const connectionString = `Authorization=${authorization};InstrumentationKey=${instrumentationKey};IngestionEndpoint=${ingestionEndpoint};LiveEndpoint=${liveEndpoint}`;
+
+            const result = ConnectionStringParser.parse(connectionString);
+
+            assert.deepEqual(result.authorization, authorization);
+            assert.deepEqual(result.instrumentationkey, instrumentationKey);
+            assert.deepEqual(result.ingestionendpoint, ingestionEndpoint);
+            assert.deepEqual(result.liveendpoint, liveEndpoint);
+        });
+
+        it("should ignore invalid fields", () => {
+            const authorization = "ikey"
+            const instrumentationKey = "ikey";
+            const ingestionEndpoint = "ingest";
+            const liveEndpoint = "live";
+            const connectionString = `Autho.rization=${authorization};Instrume.ntationKey=${instrumentationKey};Ingestion.Endpoint=${ingestionEndpoint};LiveEnd.point=${liveEndpoint}`;
+
+            const result = ConnectionStringParser.parse(connectionString);
+
+            assert.deepEqual(result.authorization, undefined);
+            assert.deepEqual(result.instrumentationkey, undefined);
```

Comment:
If an Ikey is not found, I would expect this to throw some kind of exception.

Reply:
Currently I'm relying on [this](https://github.com/microsoft/ApplicationInsights-node.js/blob/88b38caffaad1310ceb8242c5cae9bc40bf1e573/Library/Config.ts#L125) to show errors to users after all parsing is finished. I'm unsure on if I should show parsing errors here or where the results are actually used (or both?). For the most part errors are shown where the stuff is used.